examples/light-switch-app/silabs/src/AppTask.cpp

```diff
@@ -115,3 +208,100 @@ void AppTask::AppTaskMain(void * pvParameter)
         }
     }
 }
+
+void AppTask::ButtonEventHandler(uint8_t button, uint8_t btnAction)
+{
+    AppEvent event = {};
+    event.Handler  = AppTask::AppEventHandler;
+    if (btnAction == to_underlying(SilabsPlatform::ButtonAction::ButtonPressed))
+    {
+        event.Type = (button ? AppEvent::kEventType_ActionButtonPressed : AppEvent::kEventType_FunctionButtonPressed);
+    }
+    else
+    {
+        event.Type = (button ? AppEvent::kEventType_ActionButtonReleased : AppEvent::kEventType_FunctionButtonReleased);
+    }
+    AppTask::GetAppTask().PostEvent(&event);
+}
+
+void AppTask::AppEventHandler(AppEvent * aEvent)
+{
+    switch (aEvent->Type)
+    {
+    case AppEvent::kEventType_FunctionButtonPressed:
+        sFunctionButtonPressed = true;
+        if (sActionButtonPressed)
+        {
+            sActionButtonSuppressed = true;
+            LightSwitchMgr::GetInstance().changeStepMode();
+            ChipLogProgress(AppServer, "Step direction changed. Current Step Direction : %s",
+                            ((LightSwitchMgr::GetInstance().getStepMode() == StepModeEnum::kUp) ? "kUp" : "kDown"));
+        }
+        else
+        {
+            sIsButtonEventTriggered = true;
+            // Post button press event to BaseApplication
+            AppEvent button_event           = {};
+            button_event.Type               = AppEvent::kEventType_Button;
+            button_event.ButtonEvent.Action = static_cast<uint8_t>(SilabsPlatform::ButtonAction::ButtonPressed);
+            button_event.Handler            = BaseApplication::ButtonHandler;
+            AppTask::GetAppTask().PostEvent(&button_event);
+        }
+        break;
+    case AppEvent::kEventType_FunctionButtonReleased: {
+        sFunctionButtonPressed = false;
+        if (sIsButtonEventTriggered)
+        {
+            sIsButtonEventTriggered = false;
+            // Post button release event to BaseApplication
+            AppEvent button_event           = {};
+            button_event.Type               = AppEvent::kEventType_Button;
+            button_event.ButtonEvent.Action = static_cast<uint8_t>(SilabsPlatform::ButtonAction::ButtonReleased);
+            button_event.Handler            = BaseApplication::ButtonHandler;
+            AppTask::GetAppTask().PostEvent(&button_event);
+        }
+        break;
+    }
+    case AppEvent::kEventType_ActionButtonPressed:
+        sActionButtonPressed = true;
+        aEvent->Handler      = LightSwitchMgr::SwitchActionEventHandler;
```

Comment:
Should we be posting this event for all the `SwitchActionEventHandler` in this `AppEventHandler`?
We are already in the event queue handler. Why could we not call `SwitchActionEventHandler` directly rather than post another event to the queue ?